lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -3706,6 +3706,75 @@ bool OptimizeReshape::run(Function *F, const CompilationContext &cctx) {
   return changed;
 }
 
+/// Optimize resize nodes.
+bool OptimizeResize::run(Function *F, const CompilationContext &cctx) {
+  LOG_SCOPE(F->getLogContext(), getName());
+  bool changed = false;
+  // Optimize ResizeNearest node.
+  for (auto &node : F->getNodes()) {
+    auto *resizeNode = dyn_cast<ResizeNearestNode>(&node);
+    if (!resizeNode) {
+      continue;
+    }
+    auto inpDims = resizeNode->getInput().dims();
+    auto outDims = resizeNode->getResult().dims();
+    // Remove identity resize (same input and output shape).
+    if (inpDims == outDims) {
+      resizeNode->getResult().replaceAllUsesOfWith(resizeNode->getInput());
+      changed = true;
+      continue;
+    }
+    // Dimensions which are resized from unitary sizes should use Tile nodes.
+    NodeValue newOut = resizeNode->getInput();
+    for (size_t idx = 0; idx < inpDims.size(); idx++) {
+      if ((outDims[idx] > inpDims[idx]) && (inpDims[idx] == 1)) {
+        unsigned_t numTiles = outDims[idx];
+        newOut = F->createTile(node.getName().str() + "." + std::to_string(idx),
+                               newOut, numTiles, idx);
+      }
+    }
+    if (newOut != resizeNode->getInput()) {
+      newOut = F->createResizeNearest(node.getName(), newOut,
+                                      resizeNode->getResult().getType());
+      resizeNode->getResult().replaceAllUsesOfWith(newOut);
+      changed = true;
+      continue;
+    }
+  }
+  // Optimize ResizeBilinear node.
```

Comment:
Why do you need to repeat this process here?

Reply:
The logic is repeated for `ResizeNearest` and `ResizeBilinear` nodes.

Reply:
> The logic is repeated for `ResizeNearest` and `ResizeBilinear` nodes.

Okay can we make a wrapper for this process, instead of copy & paste? That should make code look cleaner, thanks

Reply:
I refactored the code to use the wrapper as you suggested. Please take a look.